apps/studio/src/components/TabQueryEditor.vue

```diff
@@ -744,51 +736,11 @@
             runQueryKeyMap["Esc"] = this.cancelQuery
           }
 
-          const modes = {
-            'mysql': 'text/x-mysql',
-            'postgresql': 'text/x-pgsql',
-            'sqlserver': 'text/x-mssql',
-            'mariadb': 'text/x-mariadb',
-            'sqlite': 'text/x-sqlite',
-            'cassandra': 'text/x-cassandra',
-            'redshift': 'text/x-pgsql',
-          };
-
-          const extraKeys = {}
-
-          extraKeys[this.cmCtrlOrCmd('F')] = 'findPersistent'
-          extraKeys[this.cmCtrlOrCmd('R')] = 'replace'
-          extraKeys[this.cmCtrlOrCmd('Shift-R')] = 'replaceAll'
-
-
-          this.editor = CodeMirror.fromTextArea(this.$refs.editor, {
-            lineNumbers: true,
-            mode: this.connection.connectionType in modes ? modes[this.connection.connectionType] : "text/x-sql",
-            tabSize: 2,
-            theme: 'monokai',
-            extraKeys: {
-              "Ctrl-Space": "autocomplete",
-              "Shift-Tab": "indentLess",
-              ...extraKeys
-            },
-            options: {
-              closeOnBlur: false
-            },
-            // eslint-disable-next-line
-            // @ts-ignore
-            hint: CodeMirror.hint.sql,
-            hintOptions: this.hintOptions,
-            keyMap: this.userKeymap,
-            getColumns: this.getColumnsForAutocomplete
-          } as CodeMirror.EditorConfiguration)
+          this.editor = this.$refs.textEditor.initialize()
           this.editor.setValue(startingValue)
           this.editor.addKeyMap(runQueryKeyMap)
-          this.editor.on("keydown", (_cm, e) => {
-            if (this.$store.state.menuActive) {
-              e.preventDefault()
-            }
-          })
-
+          this.editor.setOption('hintOptions', this.hintOptions)
```

Comment:
Again, we should not be calling methods on child components, this breaks the Vue.js model of component communication.

Reply:
Oh that's a great idea!

I initially put `hintOptions` as a prop but it's only used for codemirror so the separate component probably would look something like this:

```vue
<sql-text-editor :tables="..." @getColumns="..." />
```